<a href="https://colab.research.google.com/github/TitanVP10/ActividadFormativaM2-Equipo3/blob/main/M2_1_EthanVerduzco.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Actividad M2.1 Uso de Google Colab

# Actividad Git-GitHub

# Equipo 3

# Ethan Enrique Verduzco Pérez | A01066955


# Importing Libraries

In [ ]:
import pandas as pd 

# Lectura de Datos

In [ ]:
from google.colab import drive 
drive.mount("/content/drive")

Mounted at /content/drive


In [ ]:
#Buscar en la parte izquierda de los archivos el archivo y buscar su ruta de acceso

titanic = pd.read_csv('/content/drive/MyDrive/Ciencia de Datos/titanic(1).csv')

In [ ]:
titanic.head(10)

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
4            5         0       3  ...   8.0500   NaN         S
5            6         0       3  ...   8.4583   NaN         Q
6            7         0       1  ...  51.8625   E46         S
7            8         0       3  ...  21.0750   NaN         S
8            9         1       3  ...  11.1333   NaN         S
9           10         1       2  ...  30.0708   NaN         C

[10 rows x 12 columns]

In [ ]:
titanic.tail(10)

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
881          882         0       3  ...   7.8958   NaN         S
882          883         0       3  ...  10.5167   NaN         S
883          884         0       2  ...  10.5000   NaN         S
884          885         0       3  ...   7.0500   NaN         S
885          886         0       3  ...  29.1250   NaN         Q
886          887         0       2  ...  13.0000   NaN         S
887          888         1       1  ...  30.0000   B42         S
888          889         0       3  ...  23.4500   NaN         S
889          890         1       1  ...  30.0000  C148         C
890          891         0       3  ...   7.7500   NaN         Q

[10 rows x 12 columns]

# Descripción de Variables

In [ ]:
#Explicación de la función
#Muestra el nombre de cada una de las columnas en el Data Frame ordenadas de
#acuerdo a su aparición
titanic.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [ ]:
titanic.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

Las variables categóricas son:

1.   PassengerId
2.   Pclass
3.   Name
4.   Survived
5.   Sex
6.   Ticket
7.   Cabin
8.   Embarked

Las variables numéricas son:

1.   Age
2.   SibSp
3.   Parch
4.   Fare


In [ ]:
#Explicación de la función
#Muestra las columnas, con su respectivo número de registros No Nulos que cuenta
# y el tipo de datos que maneja 
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


**Las columnas con valores nulos son:**

1. Age (177 valores nulos)
2. Cabin (687 valores nulos)
3. Embarked (2 valores nulos)

**Las columnas con valores No nulos son:**

1. PassengerId
2. Survives
3. Pclass
4. Name
5. Sex
6. SibSp
7. Parch
8. Ticket
9. Fare


In [ ]:
#Explicación de la función
#Muestra los principales atributos estadísticos de cada columna, como su número 
#de registros totales, media, desviación estándar, mínimo, máximo y sus percentiles
titanic.describe()

PassengerId    Survived      Pclass  ...       SibSp       Parch        Fare
count   891.000000  891.000000  891.000000  ...  891.000000  891.000000  891.000000
mean    446.000000    0.383838    2.308642  ...    0.523008    0.381594   32.204208
std     257.353842    0.486592    0.836071  ...    1.102743    0.806057   49.693429
min       1.000000    0.000000    1.000000  ...    0.000000    0.000000    0.000000
25%     223.500000    0.000000    2.000000  ...    0.000000    0.000000    7.910400
50%     446.000000    0.000000    3.000000  ...    0.000000    0.000000   14.454200
75%     668.500000    1.000000    3.000000  ...    1.000000    0.000000   31.000000
max     891.000000    1.000000    3.000000  ...    8.000000    6.000000  512.329200

[8 rows x 7 columns]

Indica cuántos registros se tienen en total

Se tiene en total **891** registros





¿Qué columnas aparecen en el resultado de describe?

Las columnas que aparecen en el 'describe' son las que tienen datos de tipo 
numérico


¿Cuál es la tarifa más cara?

La tarifa más cara es de **512.3292**

¿Cuál es el promedio de edad?

**29.699118**

In [ ]:
valores_diferentes_PassengerId = titanic["PassengerId"].unique()

print('El número de valores distintos en la categoría de PassengerId son:', len(valores_diferentes_PassengerId))

El número de valores distintos en la categoría de PassengerId son: 891


In [ ]:
valores_diferentes_Pclass = titanic["Pclass"].unique()

print('El número de valores distintos en la categoría de Pclass son:', len(valores_diferentes_Pclass))

El número de valores distintos en la categoría de Pclass son: 3


In [ ]:
valores_diferentes_Name = titanic["Name"].unique()

print('El número de valores distintos en la categoría de Name son:', len(valores_diferentes_Name))

El número de valores distintos en la categoría de Name son: 891


In [ ]:
valores_diferentes_Survived = titanic["Survived"].unique()

print('El número de valores distintos en la categoría de Survived son:', len(valores_diferentes_Survived))

El número de valores distintos en la categoría de Survived son: 2


In [ ]:
valores_diferentes_Sex = titanic["Sex"].unique()

print('El número de valores distintos en la categoría de Sex son:', len(valores_diferentes_Sex))

El número de valores distintos en la categoría de Sex son: 2


In [ ]:
valores_diferentes_Ticket = titanic["Ticket"].unique()

print('El número de valores distintos en la categoría de Ticket son:', len(valores_diferentes_Ticket))

El número de valores distintos en la categoría de Ticket son: 681


In [ ]:
valores_diferentes_Cabin = titanic["Cabin"].unique()

print('El número de valores distintos en la categoría de Cabin son:', len(valores_diferentes_Cabin))

El número de valores distintos en la categoría de Cabin son: 148


In [ ]:
valores_diferentes_Embarked = titanic["Embarked"].unique()

print('El número de valores distintos en la categoría de Embarked son:', len(valores_diferentes_Embarked))

El número de valores distintos en la categoría de Embarked son: 4


# Análisis de Datos

In [ ]:
# ¿cuánto es el total que se pagó sumando la tarifa de todos los pasajeros?
total_Tarifas_pasajeros = sum(titanic['Fare'])

total_Tarifas_pasajeros

28693.949299999967

In [ ]:
#Crea un subconjunto de los datos de los que sobrevivieron
sobrevivientes = titanic[titanic['Survived'] == 1]

#Mostrar los datos de 5 personas de las que sobrevivieron
sobrevivientes.head(5)

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
8            9         1       3  ...  11.1333   NaN         S
9           10         1       2  ...  30.0708   NaN         C

[5 rows x 12 columns]

In [ ]:
#Crea un subconjunto de los datos de los que NO sobrevivieron
fallecidos = titanic[titanic['Survived'] == 0]

#Mostrar los datos de 5 personas de las que NO sobrevivieron
fallecidos.head(5)

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
4            5         0       3  ...   8.0500   NaN         S
5            6         0       3  ...   8.4583   NaN         Q
6            7         0       1  ...  51.8625   E46         S
7            8         0       3  ...  21.0750   NaN         S

[5 rows x 12 columns]